src/app/shared/toolbar/toolbar.component.html

```diff
@@ -1,75 +1,42 @@
-      />
-    </div>
-
+<mat-toolbar color="primary">
+  <div class="toolbar-content">
     <div>
-    <a mat-button href="https://github.com/flathub/flathub/wiki/App-Submission" aria-label="Publish an app on Flathub">
-      PUBLISH
-    </a>
-    <a mat-button routerLink="about">ABOUT</a>
-  </div>
-  </div>
-
-  <div fxShow fxHide.gt-sm class="toolbar-content">
+      <a class="store-logo-button" routerLink="/" aria-label="Flathub" mat-button>
+        <img src="../../../assets/themes/flathub/flathub-logo-toolbar.svg" alt="flathub">
+      </a>
+    </div>
 
-    <button *ngIf="!showSearchInput" mat-icon-button [matMenuTriggerFor]="menu" xPosition="after">
-      <mat-icon>more_vert</mat-icon>
-    </button>
+    <div class="flex-fill-remaining-space">
+      <div class="search-box-container">
+        <div [class]="'search-form '+(searchActive? 'active': '')">
+          <mat-icon class="search-box-icon show-md" (click)="searchActive=!searchActive">arrow_back</mat-icon>
+          <mat-icon class="search-box-icon hide-md">search</mat-icon>
+          <input #searchBox id="search-box" class="search-box" type="search" placeholder="Search apps" (keyup.enter)="onSearch(searchBox.value)" />
+        </div>
+      </div>
+    </div>
 
-    <mat-menu #menu="matMenu">
+    <div class="hide-md toolbar-nav">
+      <button mat-button routerLink="apps">APPLICATIONS</button>
+      <button mat-button href="https://github.com/flathub/flathub/wiki/App-Submission">PUBLISH</button>
+      <button mat-button routerLink="about">ABOUT</button>
+    </div>
 
-      <button mat-menu-item routerLink="apps">
-        <span>APPLICATIONS</span>
+    <div class="show-md">
+      <button mat-icon-button (click)="searchActive=!searchActive">
+        <mat-icon>search</mat-icon>
       </button>
-      <a mat-menu-item href="https://github.com/flathub/flathub/wiki/App-Submission" aria-label="Publish your app on Flathub">PUBLISH
-      </a>
-      <button mat-menu-item routerLink="about">
-        <span>ABOUT</span>
+      <button mat-icon-button [matMenuTriggerFor]="menu">
+        <mat-icon>more_vert</mat-icon>
       </button>
-
-    </mat-menu>
-
-    <a *ngIf="!showSearchInput" mat-button class="store-logo-button flex-fill-remaining-space" routerLink="/" aria-label="Flathub">
-      <img class="store-logo" src="../../../assets/themes/flathub/flathub-logo-toolbar.svg" alt="flathub">
-    </a>
-
-    <div *ngIf="showSearchInput" class="search-box-container flex-fill-remaining-space">
-      <mat-icon class="search-box-icon">search</mat-icon>
-
-      <input #searchBox id="search-box" class="search-box-mobile" type="search" placeholder="Search apps" (keyup.enter)="onSearch(searchBox.value)"
-      />
+      <mat-menu #menu="matMenu" overlapTrigger="false">
+        <button mat-menu-item routerLink="apps">APPLICATIONS</button>
+        <button mat-menu-item href="https://github.com/flathub/flathub/wiki/App-Submission">PUBLISH</button>
```

Comment:
External links don't work with button. The code should be like this:

` <a mat-button href="https://github.com/flathub/flathub/wiki/App-Submission" aria-label="Publish an app on Flathub">PUBLISH</a>`